### PR TITLE
Add BLE Beacon example

### PR DESCRIPTION
**Pull request by kalkyl**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,7 +36,7 @@ Republished without changes to fix missing README.md in crates.io.
 - Fix rustdoc warnings
 - Upgrade nrf51-hal to 0.12.1
 
-[Unreleased]: https://github.com/nrf-rs/microbit/compare/v0.10.1...HEAD
+[unreleased]: https://github.com/nrf-rs/microbit/compare/v0.10.1...HEAD
```
Comment (therealprof) [inline CHANGELOG.md:39]: That seems wrong, the link is really using Unreleased as anchor.

File: examples/ble-beacon/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+edition = "2018"
+
+[dependencies]
+embedded-hal = "0.2.4"
+cortex-m-rt = "0.6.12"
+panic-halt = "0.2.0"
+defmt-rtt = "0.2.0"
+defmt = "0.2.0"
+rubble = "0.0.4"
+rubble-nrf5x = { version = "0.0.4", default-features = false, features = ["51"] }
```
Comment (therealprof) [inline examples/ble-beacon/Cargo.toml:13]: I guess this needs to depend on `v1` vs`v2`?

File: examples/ble-beacon/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            )
+            .unwrap();
+            beacon.broadcast(&mut radio);
+            timer.delay_ms(500_u16);
+        }
+    }
+
+    loop {
+        continue;
+    }
```
Comment (jonas-schievink) [inline examples/ble-beacon/src/main.rs:63]:
```suggestion
    loop {}
```

File: CHANGELOG.md (hunk 1 above)
Comment (kalkyl) [reply CHANGELOG.md]: Autoformatting...arrgh

File: examples/ble-beacon/src/main.rs (hunk 3 above)
Comment (kalkyl) [reply examples/ble-beacon/src/main.rs]: Right, old habit :P

**Comment by therealprof** [comment]
Works great. I think we can apply this, unless you want to clean it up first.

**Comment by kalkyl** [comment]
Great! Let's go!

**Review by therealprof — APPROVED** [review]
LGTM, thanks.

bors r+

**Pull request merged**